library/Solarium/Core/Client/Adapter/Curl.php

```diff
@@ -256,4 +261,36 @@ protected function createOptions($request, $endpoint)
         return $options;
         // @codeCoverageIgnoreEnd
     }
+
+
+    /**
+     * Extract response headers from cURL response. The headers are embeded in
+     * the body and the section is ending with \r\n\r\n. Example:
+     * --------------------------------------------
+     *  HTTP/1.1 200 OK
+     *  Content-Type: text/plain; charset=utf-8
+     *
+     *  { ... response JSON or XML ... }
+     * --------------------------------------------
+     *
+     * @param $content
+     * @return array
+     */
+    protected function extractResponseHeaders($content)
+    {
+        $headers = array();
+        // Split the string on every "double" new line.
+        $arrRequests = explode("\r\n\r\n", $content);
+        // Loop of response headers. The "count() -1" is to
+        //avoid an empty row for the extra line break before the body of the response.
+        for ($index = 0; $index < count($arrRequests) -1; $index++) {
+            foreach (explode("\r\n", $arrRequests[$index]) as $i => $line) {
+                if ($i !== 0) { // First is HTTP 1.0 200 OK.
+                    list ($key, $value) = explode(': ', $line);
```

Comment:
It's not clear how the returned array should look like. Looking at acquia_connector, it appears that is assuming to be just strings: http://cgit.drupalcode.org/acquia_connector/tree/acquia_search/src/EventSubscriber/SearchSubscriber.php?h=8.x-1.x#n97

Do you have any idea?

```php
$headers = array('Expires:Fri, 01 Apr 2016 20:41:50 GMT', 'Keep-Alive:timeout=2, max=100')
``` 

or 

```php
$headers = array('Expires' => 'Fri, 01 Apr 2016 20:41:50 GMT', 'Keep-Alive'  => 'timeout=2, max=100');
```